sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java

```diff
@@ -0,0 +1,137 @@
+     */
+    private static final Pattern STATUS_CODE_PATTERN = Pattern
+        .compile("HTTP\\/\\d\\.\\d\\s?(\\d+)\\s?\\w+", Pattern.CASE_INSENSITIVE);
+
+    // This method connects the batch response values to the individual batch operations based on their Content-Id
+    static Mono<SimpleResponse<Void>> mapBatchResponse(BlobBatch batch, ServicesSubmitBatchResponse batchResponse,
+        boolean throwOnAnyFailure, ClientLogger logger) {
+        /*
+         * Content-Type will contain the boundary for each batch response. The expected format is:
+         * "Content-Type: multipart/mixed; boundary=batchresponse_66925647-d0cb-4109-b6d3-28efe3e1e5ed"
+         */
+        String contentType = batchResponse.getDeserializedHeaders().getContentType();
+
+        // Split on the boundary [ "multipart/mixed; boundary", "batchresponse_66925647-d0cb-4109-b6d3-28efe3e1e5ed"]
+        String boundary = contentType.split("=", 2)[1];
+
+        return FluxUtil.collectBytesInByteBufferStream(batchResponse.getValue())
+            .flatMap(byteArrayBody -> {
+                String body = new String(byteArrayBody, StandardCharsets.UTF_8);
+
+                // Split the batch response body into batch operation responses.
+                for (String subResponse : body.split("--" + boundary)) {
+                    // This is a split value that isn't a response.
+                    if (!subResponse.contains("application/http")) {
+                        continue;
+                    }
+
+                    // The batch operation response will be delimited by two new lines.
+                    String[] subResponseSections = subResponse.split(HTTP_NEWLINE + HTTP_NEWLINE);
+
+                    // The first section will contain batching metadata.
+                    BlobBatchOperationResponse<?> batchOperationResponse =
+                        getBatchOperation(batch, subResponseSections[0], logger);
+
+                    // The second section will contain status code and header information.
+                    setStatusCodeAndHeaders(batchOperationResponse, subResponseSections[1]);
+
+                    // The third section will contain the body.
+                    if (subResponseSections.length > 2) {
+                        // The body is optional and may not exist.
+                        setBodyOrPotentiallyThrow(batchOperationResponse, subResponseSections[2], throwOnAnyFailure,
+                            logger);
+                    }
+                }
+
+                return Mono.just(new SimpleResponse<>(batchResponse, null));
+            });
+    }
+
+    private static BlobBatchOperationResponse<?> getBatchOperation(BlobBatch batch, String responseBatchInfo,
+        ClientLogger logger) {
+        Matcher contentIdMatcher = CONTENT_ID_PATTERN.matcher(responseBatchInfo);
+
+        int contentId;
+        if (contentIdMatcher.find()) {
+            contentId = Integer.parseInt(contentIdMatcher.group(1));
```

Comment:
The regex you are using, actually matches the group optionally, so this group(1) you're assuming exists here can fail.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/5756